api/src/services/verification-service.ts

```diff
@@ -242,4 +216,26 @@ export class VerificationService {
 		await this.userService.updateUserVerification(vup);
 		await this.userService.addUserVC(vc);
 	};
+
+	getKeyCollectionIndex = (currentCredentialIndex: number) => Math.floor(currentCredentialIndex / KEY_COLLECTION_SIZE);
+
+	private generateKeyCollection = async (
```

Comment:
I would probably wrap the body of this function with try/catch, just to avoid a possible `Unhandled Promise rejection` error.
Also I would add try/catch for `getKeyCollection` function

Reply:
I will add try catches around the functions called by the routes to avoid this, is this the correct approach?

Reply:
alright, sounds good!